presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -581,7 +581,7 @@ private Map<String, HiveColumnStatistics> groupStatisticsByColumn(List<ColumnSta
     }
 
     @Override
-    public synchronized void updateTableStatistics(MetastoreContext metastoreContext, String databaseName, String tableName, Function<PartitionStatistics, PartitionStatistics> update)
+    public void updateTableStatistics(MetastoreContext metastoreContext, String databaseName, String tableName, Function<PartitionStatistics, PartitionStatistics> update)
```

Comment:
This might be just a temp solution. I think the more correct way is to 
1. Separate the locks on each partition and column;
2. Separate the shared and exclusive locks. E.g. getPartitionColumnStatistics gets the shared lock on the column instead of an exclusive lock.

But since other methods like setTableColumnStatistics, deleteTableColumnStatistics are also not synchronized, I guess we can merge this one first to unblock people, and do the modification in the future.

@arhimondr what do you think?

Reply:
@yingsu00 Yes, the current solution should still have problems in the case of concurrency. To solve this problem, I think it is necessary to introduce ACID of Hive metastore to solve it.